src/test/java/edu/kit/provideq/toolbox/api/SharpSatSolversTest.java

```diff
@@ -0,0 +1,64 @@
+package edu.kit.provideq.toolbox.api;
+
+
+import static edu.kit.provideq.toolbox.sharpsat.SharpSatConfiguration.SHARPSAT;
+
+import edu.kit.provideq.toolbox.meta.ProblemManagerProvider;
+import edu.kit.provideq.toolbox.meta.ProblemSolver;
+import java.time.Duration;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest
+@AutoConfigureMockMvc
+class SharpSatSolversTest {
+  @Autowired
+  private WebTestClient client;
+
+  @Autowired
+  private ProblemManagerProvider problemManagerProvider;
+
+
+  @BeforeEach
+  void beforeEach() {
+    this.client = this.client.mutate()
+        .responseTimeout(Duration.ofSeconds(20))
+        .build();
+  }
+
+  @SuppressWarnings("OptionalGetWithoutIsPresent")
+  Stream<Arguments> provideArguments() {
+    var problemManager = problemManagerProvider.findProblemManagerForType(SHARPSAT).get();
+
+    return ApiTestHelper.getAllArgumentCombinations(problemManager)
+        .map(list -> Arguments.of(list.get(0), list.get(1)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideArguments")
+  void testSatSolver(ProblemSolver<String, Integer> solver, String input) {
+    var problem = ApiTestHelper.createProblem(client, solver, input, SHARPSAT);
+    ApiTestHelper.testSolution(problem);
+  }
+
```

Comment:
Minor note, could you remove the empty lines here? (+ some other files also have empty lines at the end of the class)